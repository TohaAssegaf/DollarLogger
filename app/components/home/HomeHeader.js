import styles from './styles'
import * as PaymentUtils from '~/app/lib/PaymentUtils'
import * as MoneyFormatter from '~/app/lib/MoneyFormatter'
import React from 'react'
import { Text, View } from 'react-native'
import { connect } from 'react-redux'

class HomeHeader extends React.Component {
  getCurrentSpend() {
    return PaymentUtils.getTotalSpend(PaymentUtils.filterCurrentWeekPayments(this.props.payments))
  }

  getRemainingBudget() {
    return this.props.budgetTotal - this.getCurrentSpend()
  }

  render() {
    return (
      <View style={styles.header}>
        <View style={styles.cell}>
          <Text style={styles.amount}>
            {MoneyFormatter.formatMoney(this.getRemainingBudget())}
          </Text>
          <Text>Remaining this week</Text>
        </View>
        <View style={styles.cell}>
          <Text style={styles.amount}>
            {MoneyFormatter.formatMoney(this.getCurrentSpend())}
          </Text>
          <Text>Spent so far</Text>
        </View>
      </View>
    );
  }
}

const mapStateToProps = state => {
  return {
    budgetTotal: state.budget.total,
    payments: state.payment.payments
  }
}

export default connect(mapStateToProps)(HomeHeader)
