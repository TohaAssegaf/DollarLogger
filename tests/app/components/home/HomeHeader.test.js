import HomeHeader from '/app/components/home/HomeHeader'
import React from 'react';
import configureMockStore from 'redux-mock-store'
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

it('renders correctly', () => {
  const store = mockStore({ budget: { total: 25000 }, payment: { payments: [] }})
  const rendered = renderer.create(<HomeHeader store={store} />).toJSON();
  expect(rendered).toMatchSnapshot();
});
