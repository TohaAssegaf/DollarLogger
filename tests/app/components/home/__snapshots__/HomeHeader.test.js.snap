// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#222222",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.5,
        "flexDirection": "column",
        "height": 100,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 32,
        }
      }
    >
      $
      250.00
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Remaining this week
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.5,
        "flexDirection": "column",
        "height": 100,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 32,
        }
      }
    >
      $
      0.00
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Spent so far
    </Text>
  </View>
</View>
`;
